Lib/posixpath.py

```diff
@@ -474,26 +475,22 @@ def realpath(filename, *, strict=False):
             if not stat.S_ISLNK(st.st_mode):
                 path = newpath
                 continue
+            if newpath in seen:
+                # Already seen this path
+                path = seen[newpath]
+                if path is not None:
+                    # use cached value
+                    continue
+                # The symlink is not resolved, so we must have a symlink loop.
+                raise OSError(errno.ELOOP, "Symlink loop", newpath)
```

Comment:
Why not keep `os.stat(newpath)` here?

Reply:
For consistency with the way we raise `NotADirectoryError` in #118290, and to save the small cost of a `stat()`

Reply:
Well, there's no point in being consistent with something that hasn't landed yet, so I'll switch back to `os.stat(newpath)`. Thanks for your suggestion.